post.js

```diff
@@ -0,0 +1,163 @@
+async function loadPost(url) {
+  try {
+    const response = await fetch(url);
+    if (!response.ok) {
+      throw new Error(`HTTP error! Status: ${response.status}`);
+    }
+    return await response.text();
+  } catch (error) {
+    console.error("Error fetching post:", error);
+  }
+}
+
+async function parsePostHtml(htmlContent) {
+  const parser = new DOMParser();
+  const doc = parser.parseFromString(htmlContent, "text/html");
+  const alternateLink = doc.querySelector('link[rel="alternate"]');
+  return alternateLink ? alternateLink.href : null;
+}
+
+async function fetchJsonLd(jsonLdUrl) {
+  try {
+    const response = await fetch(jsonLdUrl);
+    if (!response.ok) {
+      throw new Error(`HTTP error! Status: ${response.status}`);
+    }
+    return await response.json();
+  } catch (error) {
+    console.error("Error fetching JSON-LD:", error);
+  }
+}
+
+function renderPost(jsonLdData) {
+  const postContainer = document.getElementById("post-container");
+  let contentHtml = "";
```

Comment:
Instead of constructing a string we should be making use of DOM APIs, otherwise we risk introducing [XSS vulnerabilities](https://en.wikipedia.org/wiki/Cross-site_scripting)

Reply:
Rendering has been moved to a method in the DistributedPost component, and that it now safely uses DOM APIs for content construction.